win32ss/user/user32/windows/cursoricon.c

```diff
@@ -2025,13 +2025,17 @@ HANDLE WINAPI CopyImage(
 {
     TRACE("hImage=%p, uType=%u, cxDesired=%d, cyDesired=%d, fuFlags=%x\n",
         hImage, uType, cxDesired, cyDesired, fuFlags);
+    HANDLE handle;
     switch(uType)
     {
         case IMAGE_BITMAP:
             return BITMAP_CopyImage(hImage, cxDesired, cyDesired, fuFlags);
         case IMAGE_CURSOR:
         case IMAGE_ICON:
-            return CURSORICON_CopyImage(hImage, uType == IMAGE_ICON, cxDesired, cyDesired, fuFlags);
+            handle = CURSORICON_CopyImage(hImage, uType == IMAGE_ICON, cxDesired, cyDesired, fuFlags);
+            if(handle == NULL && (fuFlags & (LR_COPYFROMRESOURCE|LR_COPYRETURNORG)))
+                handle = CURSORICON_CopyImage(hImage, uType == IMAGE_ICON, cxDesired, cyDesired, (fuFlags & ~LR_COPYFROMRESOURCE));
+            return handle;
```

Comment:
```suggestion
        {
            HANDLE handle = CURSORICON_CopyImage(hImage, uType == IMAGE_ICON, cxDesired, cyDesired, fuFlags);
            if (!handle && (fuFlags & (LR_COPYFROMRESOURCE|LR_COPYRETURNORG)))
                handle = CURSORICON_CopyImage(hImage, uType == IMAGE_ICON, cxDesired, cyDesired, (fuFlags & ~LR_COPYFROMRESOURCE));
            return handle;
        }
```

Reply:
IIRC: That first patch failed for me under GCC (is it before C99?):
```
../win32ss/user/user32/windows/cursoricon.c: In function 'CopyImage':
../win32ss/user/user32/windows/cursoricon.c:2034:13: error: a label can only be part of a statement and a declaration is not a statement
             HANDLE handle = CURSORICON_CopyImage(hImage, uType == IMAGE_ICON, cxDesired, cyDesired, fuFlags);
             ^~
```
But I have to check.

Reply:
yes that's why you need the curly brackets for the case XXX: , as in my suggestion.